packages/teleport/src/AppLauncher/useAppLauncher.ts

```diff
@@ -31,9 +32,15 @@ export default function useAppLauncher() {
         // make a redirect to the requested app auth endpoint
         const location = window.location;
         const port = location.port ? ':' + location.port : '';
-        window.location.replace(
-          `https://${result.fqdn}${port}/x-teleport-auth#value=${result.value}`
-        );
+        const state = getUrlParameter('state', location.search);
+        const authUrl = `https://${result.fqdn}${port}/x-teleport-auth`;
```

Comment:
@andrejtokarcik @alex-kovoy any reason we are not using 

https://javascript.info/url#creating-a-url

Reply:
Feel free to ignore me if I'm missing something here

Reply:
Hey @klizhentas this PR is part of a public security release I'm preparing at this very moment. I'll be able to address this only later, though I personally don't see a benefit to using `URL` in this case.

Reply:
I'm assuming URL should be safer because it can escape the parameters that contain special characters.

Reply:
That string enforces the `https://` schema which tells a browser about URL (this is to avoid `javascript://` or other stuff that might come from the server).